lib/connectors/memory.js

```diff
@@ -723,6 +725,62 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replace = function replace(model, id, data, options, cb) {
+  var self = this;
+  if (!id) {
+    var err = new Error('You must provide an id when replacing!');
+    if (cb) {
+      return cb(err);
+    } else {
+      throw err;
+    }
```

Comment:
The code below already requires `cb` to be set, I think we can do the same here.

```js
if (!id) {
  return cb(new Error('...'));
}
```